File: jdroid-android-sample/src/main/java/com/jdroid/android/sample/ui/uri/MainIntentErrorUriHandler.java
package com.jdroid.android.sample.ui.uri;

import android.app.Activity;
import android.content.Intent;
import android.net.Uri;
import androidx.annotation.NonNull;

import com.jdroid.android.uri.AbstractUriHandler;

public class MainIntentErrorUriHandler extends AbstractUriHandler<MainIntentErrorActivity> {
	@Override
	public Boolean matches(Uri uri) {
		return true;
	}

	@Override
	public Intent createMainIntent(@NonNull Activity activity, Uri uri) {
		throw new RuntimeException();
	}
}

File: jdroid-android-sample/src/main/java/com/jdroid/android/sample/ui/uri/MainIntentErrorUriHandler.kt
package com.jdroid.android.sample.ui.uri

import android.app.Activity
import android.content.Intent
import android.net.Uri

import com.jdroid.android.uri.AbstractUriHandler

class MainIntentErrorUriHandler : AbstractUriHandler<MainIntentErrorActivity>() {
    override fun matches(uri: Uri): Boolean {
        return true
    }

    override fun createMainIntent(activity: Activity, uri: Uri): Intent? {
        throw RuntimeException()
    }
}
